core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -267,16 +272,22 @@ private AggregationNode createPreAggregation(
                 Optional.empty());
     }
 
-    private ProjectNode createPreProjection(
+    private Optional<ProjectNode> createPreProjection(
             PlanNode source,
+            Assignments projectAssignments,
             Assignments groupingExpressions,
             Map<PreAggregationKey, PreAggregation> preAggregations,
             Context context)
     {
         Assignments.Builder assignments = Assignments.builder();
         assignments.putAll(groupingExpressions);
         preAggregations.values().forEach(aggregation -> assignments.put(aggregation.getProjectionSymbol(), aggregation.getProjection()));
-        return new ProjectNode(context.getIdAllocator().getNextId(), source, assignments.build());
+        Assignments newAssignments = assignments.build();
+        if (newAssignments.getMap().values().equals(projectAssignments.getMap().values())) {
```

Comment:
This is not going to be good enough for all scenarios. If the order of the assignments changes, they will be considered different by this check, even though, semantically, it's the same operations.